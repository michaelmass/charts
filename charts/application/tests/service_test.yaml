suite: test service
templates:
  - 'service.yaml'
tests:
  - it: manifest should match snapshot
    set:
      service:
        enabled: true
    asserts:
      - matchSnapshot: {}
  - it: labels & annotations should be set
    set:
      common:
        labels:
          test_common: test
        annotations:
          test_common: test
      service:
        enabled: true
        labels:
          test_service: bar
        annotations:
          test_service: bar
    asserts:
      - isKind:
          of: Service
      - isSubset:
          path: metadata.labels
          content:
            test_common: test
            test_service: bar
      - isSubset:
          path: metadata.annotations
          content:
            test_common: test
            test_service: bar
