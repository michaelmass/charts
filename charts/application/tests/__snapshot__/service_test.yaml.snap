manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: application
        helm.sh/chart: application-0.0.1
      name: RELEASE-NAME-application
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          nodePort: null
          port: 80
          protocol: TCP
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: application
      sessionAffinity: None
      type: ClusterIP
